yarn-project/acir-simulator/src/avm/opcodes/memory.test.ts

```diff
@@ -64,6 +65,27 @@ describe('Memory instructions', () => {
       expect(actual).toEqual(new Uint32(1234n));
       expect(tag).toEqual(TypeTag.UINT32);
     });
+
+    it('should correctly set value and tag (truncating)', async () => {
+      await new Set(/*indirect=*/ 0, /*inTag=*/ TypeTag.UINT16, /*value=*/ 0x12345678n, /*offset=*/ 1).execute(
```

Comment:
Based on the instruction spec in yellow paper, the value should be of type corresponding to TypeTag. At least, if serialized in this way, the truncation should never happen. I would rather return an error here.